diff --git a/Agda.cabal b/Agda.cabal
index e9afcfb..3c67fef 100644
--- a/Agda.cabal
+++ b/Agda.cabal
@@ -183,6 +183,8 @@ custom-setup
 library
   hs-source-dirs:   src/full
 
+  c-sources: src/fixes/tzset.c
+
   if flag(cpphs)
     -- We don't write an upper bound for cpphs because the
     -- `build-tools` field can not be modified in Hackage.
diff --git a/src/fixes/tzset.c b/src/fixes/tzset.c
new file mode 100644
index 0000000..623539c
--- /dev/null
+++ b/src/fixes/tzset.c
@@ -0,0 +1 @@
+int tzset() { return 0; }
